packages/legend-graph/src/models/protocols/pure/v1/engine/V1_Engine.ts

```diff
@@ -471,12 +472,16 @@ export class V1_Engine {
     generationMode: GenerationMode,
     model: V1_PureModelContextData,
   ): Promise<V1_GenerationOutput[]> {
+    const grammar = this.pureModelContextDataToPureCode(model);
+    const pureModelContextText = new V1_PureModelContextText();
+    pureModelContextText.serializer = model.serializer;
+    pureModelContextText.code = await grammar;
```

Comment:
When we do this conversion to `PureModelContextText`, could you also add a comment to explain this? I think in the line of
```
// NOTE: here instead of sending PureModelContextData, we send PureModelContextText so
// engine can convert that back to PureModelContextData to obtain source information
// as some generator uses that info. Sending PureModelContextData with source information
// from the front end to engine would take up a lot of bandwidth.
```